src/ucs/sys/string.c

```diff
@@ -108,15 +108,15 @@ uint64_t ucs_string_to_id(const char* str)
 
 void ucs_memunits_to_str(size_t value, char *buf, size_t max)
 {
-    static const char * suffixes[] = {"", "k", "m", "g", "t"};
+    static const char * suffixes[] = {"", "k", "m", "g", "t", NULL};
 
     const char **suffix;
 
     if (value == SIZE_MAX) {
         strncpy(buf, "(inf)", max);
     } else {
         suffix = &suffixes[0];
-        while ((value >= 1024) && ((value % 1024) == 0)) {
+        while ((value >= 1024) && ((value % 1024) == 0) && *(suffix + 1)) {
```

Comment:
seems like "t" suffix won't be reached with this condition... do we have gtest for this?

Reply:
nop, it is reached

Reply:
I see now.. it would be nice to have regression test

Reply:
added